brats/losses.py

```diff
@@ -34,3 +35,69 @@ def __call__(self, prediction: torch.Tensor,
             torch.Tensor: Computed Dice Loss
         """
         return torch.sum(1 - F.dice(prediction, target, self.epsilon))
+
+
+class NLLLossOneHot(nn.Module):
+    """
+    Compute negative log-likelihood loss for one hot encoded targets.
+    """
+
+    def __init__(self, *args, **kwargs):
+        """
+
+        Args:
+            *args: Arguments accepted by the torch.nn.NLLLoss() constructor
+            **kwargs: Keyword arguments accepted by the torch.nn.NLLLoss()
+                    constructor
+        """
+        super().__init__()
+        self.nll_loss = torch.nn.NLLLoss(*args, **kwargs)
+
+    def __call__(self, prediction: torch.Tensor,
+                 target: torch.Tensor) -> torch.Tensor:
+        """
+        Args:
+            prediction: output of a network, expected to be already binarized.
+                Dimensions - (Batch, Channels, Depth, Height, Width)
+            target: labels. The classes (channels) should be one-hot encoded
+                Dimensions - (Batch, Channels, Depth, Height, Width)
+        Returns:
+            torch.Tensor: Computed NLL Loss
+        """
+        target = torch.argmax(target, dim=CLASS_DIM)
+        prediction = torch.log(prediction)
+        return self.nll_loss(prediction, target)
+
+
+class DiceWithNLLOneHot(nn.Module):
```

Comment:
What do you think about something like this: 
```python
class ComposedLoss(Loss):
    def __init___(self, losses:typing.List[Loss],weights:typing.list[float]):
        usuall_crap
   def __call__(self, prediction: torch.Tensor, target: torch.Tensor) -> torch.Tensor:
        calculate_weighted_sum
```
then we would do:
```python
loss = ComposedLoss([DiceLoss(...), NLLOneHot(...)],[1.0, 1.0])
```

Loss class would be an interface for all loses

Reply:
Accepted